renku/core/management/workflow/converters/cwl.py

```diff
@@ -260,16 +262,12 @@ def _convert_step(
                 # workflow needs to create subdirectory for output file,
                 # if the directory was not already present
                 workdir_req.listing.append(
-                    cwlgen.InitialWorkDirRequirement.Dirent(
-                        entry='$({"listing": [], "class": "Directory"})', entryname=path, writable=True
-                    )
+                    cwl.Dirent(entry='$({"listing": [], "class": "Directory"})', entryname=path, writable=True)
                 )
                 dirents.append(path)
                 jsrequirement = True
 
-            environment_variables.append(
-                cwlgen.EnvVarRequirement.EnvironmentDef(f"{RENKU_ENV_PREFIX}{output_.name}", output_.actual_value)
-            )
+            environment_variables.append(cwl.EnvironmentDef(f"{RENKU_ENV_PREFIX}{output_.name}", output_.actual_value))
```

Comment:
@SwissDataScienceCenter/renku-devs i wonder why these env var definitions? same goes for inputs and parameters (see below).

Reply:
I think you just tagged the whole renku team ;)  We'd be `renku-python-maintainers`

They can be used by scripts/commands and are also how `renku.api.Parameter` picks up values set with `--set` (and in the future, `renku.api.Output` and `renku.api.Input` as well), so people can properly use those in their scripts.

Reply:
\o/ yey, for some reason github was only offering this in it's auto-completion :< sorry people

Reply:
i'm not so sure what you mean by "properly use those in their scripts". imo passing config values via env vars is a very bad practice :(

Reply:
so I have a script train.py
```
from renku.api import Parameter
learning_rate = Parameter("learning_rate", 0.1)
train_model(learning_rate)
```

And I do `renku run my-run -- python script.py`. Then `learning_rate` gets detected as a parameter, but doesn't show up on the CLI.

So if later I do `renku workflow execute --set learning_rate=50 my-run`, before that didn't do anything. We need a way for `Parameter` to pick up the modified value. But it can't be passed on the CLI, since the script doesn't have `argparse` or similar.

So the two way to do this is to write the value to a file and have `Parameter()` pick it up, or to set it as an ENV var and have it pick it up there. When I did that change, I opted for ENV vars as then we don't need to worry about littering the file system, permissions, path issues, parallelism (The env vars are per process, a file in a fixed path wouldn't be) etc.